pkg/node/node_address_linux.go

```diff
@@ -17,23 +17,27 @@
 package node
 
 import (
+	"bytes"
 	"errors"
 	"fmt"
 	"net"
+	"sort"
 
 	"golang.org/x/sys/unix"
 
 	"github.com/vishvananda/netlink"
 )
 
-func firstGlobalV4Addr(intf string) (net.IP, error) {
+// firstGlobalV4Addr  returns the first IP address for the given interface, if
+// the wishIP is send it'll return the wishIP if is in the list of address.
+func firstGlobalV4Addr(intf string, wishIP net.IP) (net.IP, error) {
```

Comment:
Can we switch `wishIP` to `preferredIP`? I realise that wishIP makes sense but it's a little ambiguous because it might mean we're allocating that IP (maybe it's clear... shrug)